### PR TITLE
DEP: deprecate scipy.stats.signaltonoise

### PR DESCRIPTION
**Pull request by aeklant**
See issue #609

### CODE REVIEW

**Comment by rgommers** [comment]
@aeklant deprecating this function makes sense and is done correctly. A couple more things to do here though:
- mention on the mailing list that this PR deprecates them, to see if there are no objections (you can include ``histogram2`` in the same email).
- add the deprecation to the release notes (``doc/release/0.16.0-notes.rst``).
- silence the ``DeprecationWarning`` that's emitted. You can use ``warnings.filterwarnings('ignore', ...`` for that inside a context manager, search the stats tests for examples.

**Comment by aeklant** [comment]
Hi @rgommers. I added the release notes for both `histogram2` and `signaltonoise` in the same commit to avoid merge issues but let me know if I should change that. I also didn't find any test cases for `histogram2` so no `DeprecationWarning` to silence there. 

I started a discussion thread in the mailing list to see if the change is well accepted or not.

Let me know if I should do a rebase of this PR to keep things tidy.

File: scipy/stats/tests/test_stats.py [hunk 1]
```diff
@@ -1263,8 +1263,10 @@ def test_signaltonoise(self):
 
         # y = stats.signaltonoise(self.shoes[0])
         # assert_approx_equal(y,4.5709967)
-        y = stats.signaltonoise(self.testcase)
-        assert_approx_equal(y,2.236067977)
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore", RuntimeWarning)
```
Comment (rgommers) [inline scipy/stats/tests/test_stats.py]: The second argument is the warning category to filter, should be ``DeprecationWarning`` in this case.

**Comment by rgommers** [comment]
The one commit to add both functions to the release notes is fine I think.

Three separate commits is still OK and doesn't need a rebase. You need one more though to change the test; then you may want to rebase and squash.

**Comment by aeklant** [comment]
@rgommers ok, done. Now we wait for community feedback I suppose.

**Comment by rgommers** [comment]
Yep. I don't expect major objections, but we need to make sure that there's not someone who happens to have a good reason to have the deprecation or removal delayed. Decision making in public is important.

**Comment by rgommers** [comment]
Also merged. Nice going Abraham.

**Comment by aeklant** [comment]
Great! Happy to contribute :)

**Pull request merged**